[Docs]:Enhanced Installation & Getting Started Docs (#5904)
All-Hands-AI/OpenHands: docs/modules/usage/installation.mdx

## System Requirements

* **Docker version 26.0.0+ or Docker Desktop 4.31.0+**. You can check your docker version by running `docker --version` in a terminal. See the [Docker documentation for installing Docker Engine](https://docs.docker.com/engine/install/) if needed.
* Supported Operation Systems
    * **Debian/Ubuntu based Linux** - these instructions are written and tested in Debian/Ubuntu Linux systems and should work as expected.

>     * **Mac OS and non-Debian Linux** - these instructions should work as expected but aren't explicitly tested on these OSes.

Reviewer: Just for clarity: many if not all maintainers use MacOS, so usage on MacOS is probably implicitly tested 😅

I think this is also a bit confusing: this should be about the OS of the user, the machine. But the docker images, both for the `app` docker (that they run with `docker run`), and for the `runtime` (sandbox) docker (which runs the agent during task execution), are by default Debian/Ubuntu based.

    * **Windows** - you must use Windows Subsystem for Linux (WSL). For more information on WSL see the [Microsoft Documentation for WSL](https://learn.microsoft.com/en-us/windows/wsl/install). If someone is able to test and document deploying OpenHands using Docker on native Windows it would be much appreciated.

## Choosing a Deployment Method

There are two ways you can deploy the OpenHands app:

> * **Using the official Docker image (recommended)** - this method is the primary method for using OpenHands. It is the simplest and doesn't require any development experience.

Reviewer: ```suggestion
* **Using the official Docker image (recommended)** - this method is the primary method for using OpenHands. It is the simplest, you just run a command.
```

> * **Building the OpenHands app from source (for developers only)** - this method is designed for developers working on the OpenHands application. It requires experience building applications from source and working with dependencies. If you only plan to use OpenHands you shouldn't use this method.

Reviewer: While this is generally quite fair, I should mention here that I'm not fully sure of the implication that it's useless unless you plan to work on OpenHands.

There is at least one reason why it's not so: OpenHands is more easily configurable if you run without the `app` docker, because the user can tweak a configuration file and go ahead, as opposed to attempting to edit a single command (as you mentioned yourself, that's not always friendly). There are also a few `config.toml` features which are not available on the `app` docker.
Nothing here requires to develop OpenHands, just to use it more easily or powerfully (once you went through `git clone` `make build` step).

## Deploying with Docker

This is the easiest way to run OpenHands and can be easily setup to run in a bash script so you can quickly start and stop the OpenHands app. 

We will run the commands below in a terminal window. First, we make a new directory called `openhands` in your home directory to store Openhands related data. We'll also create a sub-directory called `.openhands-state` that will store session history for Open Hands.

```bash
mkdir ~/openhands
mkdir ~/openhands/.openhands-state

> cd ~/openhands

Reviewer: Saving sessions in `.openhands-state` is a relatively new behavior introduced last month -ish, about the time you first tried `openhands` and I seem to recall you ran into some bug with it. I think that was a special circumstance though, and it shouldn't normally happen.

I think, please correct me if wrong, that doing this is unnecessary now?

    -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:0.17-nikolaik \
    -e JWT_SECRET=07ae43ca-7b66-4d99-857a-a9d7126cce18 \
    -e LOG_ALL_EVENTS=true \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -v ~/openhands/.openhands-state:/.openhands-state \
    -p 3000:3000 \
    --add-host host.docker.internal:host-gateway \
    --name openhands-app \
    docker.all-hands.dev/all-hands-ai/openhands:0.17
```

We're now ready to save this to a file called `start_openhands.sh` in the new `~/openhands` directory. Finally, in the terminal window, we need to add execution permissions to the new script using the `chmod` command and then run it.

```bash
chmod +x start_openhands.sh
./start_openhands.sh
```

After running the command above, you'll find OpenHands running at [http://localhost:3000](http://localhost:3000) (change the port number to your `LOCAL_PORT` value if you customised it).

To close down the OpenHands app simply type `Ctrl+C` in the terminal window or close the terminal window.

**TODO** - this is a bit nasty, is there a cleaner way to close down the app? Does a hard stop risk corrupting things liek session state?

> 

Reviewer: Yes, there's a risk that it does not save _some_ of the state. The previous events are there, but whether you can always restore them neatly is not obvious. This might be worth its own separate issue, so we can test the current behavior in all cases here, and take it from there.
